dialdbg/src/parse.rs

```diff
@@ -0,0 +1,335 @@
+use anyhow::{bail, Result};
+use chrono::{DateTime, Duration, FixedOffset};
+use std::{fmt, fs, io, net::SocketAddr, path::PathBuf};
+use viam::rpc::log_prefixes;
+
+const DEVELOPMENT: Option<&'static str> = option_env!("DIALDBG_DEVELOPMENT");
+
+// This prefix is prepended in dialdbg when connect returns an error. It is not
+// from dial itself.
+pub(crate) const DIAL_ERROR_PREFIX: &'static str = "unexpected dial connect error";
+
+#[derive(Debug, Default)]
+pub(crate) struct GRPCResult {
```

Comment:
Probably over-engineering right now, but it would be neat if we could dedup some of the shared code across `GRPCResult` and `WebRTCResult`

Reply:
That's a good point; I probably won't in this PR, but I can imagine deduping more in the future as the code around these structs increases (potentially as part of [RSDK-3444](https://viam.atlassian.net/browse/RSDK-3444)).